packages/wkt/test/wkt.test.js

```diff
@@ -604,3 +604,434 @@ test('should parse a MULTIPOLYGON with a hole', function (t) {
+          ]
+        ],
+        properties: {
+          z: true
+        }
+      }
+    ],
+    properties: {
+      z: true
+    }
+  });
+});
+
+test('should parse a GEOMETRYCOLLECTION with M property', function (t) {
+  t.plan(1);
+
+  const input = 'GEOMETRYCOLLECTION M ( POINT M (40 10 15), LINESTRING M (10 10 15, 20 20 15, 10 40 15), POLYGON M ((40 40 15, 20 45 15, 45 30 15, 40 40 15)))';
+
+  const output = wktToGeoJSON(input);
+
+  t.deepEqual(output, {
+    type: 'GeometryCollection',
+    geometries: [
+      {
+        type: 'Point',
+        coordinates: [40, 10, 15],
+        properties: {
+          m: true
+        }
+      },
+      {
+        type: 'LineString',
+        coordinates: [
+          [10, 10, 15],
+          [20, 20, 15],
+          [10, 40, 15]
+        ],
+        properties: {
+          m: true
+        }
+      },
+      {
+        type: 'Polygon',
+        coordinates: [
+          [
+            [40, 40, 15],
+            [20, 45, 15],
+            [45, 30, 15],
+            [40, 40, 15]
+          ]
+        ],
+        properties: {
+          m: true
+        }
+      }
+    ],
+    properties: {
+      m: true
+    }
+  });
+});
+
+test('should parse a GEOMETRYCOLLECTION with geometry collection embedded', function (t) {
+  t.plan(1);
+
+  const input = 'GEOMETRYCOLLECTION ( POINT (40 10), LINESTRING (10 10, 20 20, 10 40), POLYGON ((40 40, 20 45, 45 30, 40 40)), GEOMETRYCOLLECTION ( POINT (40 10), LINESTRING (10 10, 20 20, 10 40), POLYGON ((40 40, 20 45, 45 30, 40 40))))';
+
+  const output = wktToGeoJSON(input);
+
+  t.deepEqual(output, {
+    type: 'GeometryCollection',
+    geometries: [
+      {
+        type: 'Point',
+        coordinates: [40, 10]
+      },
+      {
+        type: 'LineString',
+        coordinates: [
+          [10, 10],
+          [20, 20],
+          [10, 40]
+        ]
+      },
+      {
+        type: 'Polygon',
+        coordinates: [
+          [
+            [40, 40],
+            [20, 45],
+            [45, 30],
+            [40, 40]
+          ]
+        ]
+      },
+      {
+        type: 'GeometryCollection',
+        geometries: [
+          {
+            type: 'Point',
+            coordinates: [40, 10]
+          },
+          {
+            type: 'LineString',
+            coordinates: [
+              [10, 10],
+              [20, 20],
+              [10, 40]
+            ]
+          },
+          {
+            type: 'Polygon',
+            coordinates: [
+              [
+                [40, 40],
+                [20, 45],
+                [45, 30],
+                [40, 40]
+              ]
+            ]
+          }
+        ]
+      }
+    ]
+  });
+});
+
+test('should parse a GEOMETRYCOLLECTION with ZM property', function (t) {
+  t.plan(1);
+
+  const input = 'GEOMETRYCOLLECTION ZM ( POINT ZM (40 10 10 15), LINESTRING ZM (10 10 10 15, 20 20 10 15, 10 40 10 15), POLYGON ZM ((40 40 10 15, 20 45 10 15, 45 30 10 15, 40 40 10 15)))';
```

Comment:
this is much better!
PRs that include comprehensive tests are a maintainers dream ☁️🌈